src/file/FileAppendTransaction.js

```diff
@@ -445,6 +457,90 @@ export default class FileAppendTransaction extends Transaction {
         return "fileAppend";
     }
 
+    /**
+     * Build all the transactions when transacgtions are not signed
+     * @override
+     * @internal
+     */
+    _buildIncompletedTransactions() {
+        if (this._contents == null) {
+            throw new Error("contents is not set");
+        }
+        if (this.maxChunks && this.getRequiredChunks() > this.maxChunks) {
+            throw new Error(
+                `cannot build \`FileAppendTransaction\` with more than ${this.maxChunks} chunks`,
+            );
+        }
+
+        let nextTransactionId = TransactionId.withValidStart(
+            new AccountId(0, 0, 0),
+            new Timestamp(new Date().getTime(), 0),
+        );
+
+        // Hack around the locked list. Should refactor a bit to remove such code
+        this._transactionIds.locked = false;
+
+        this._transactions.clear();
+        this._transactionIds.clear();
+        this._signedTransactions.clear();
+
+        for (let chunk = 0; chunk < this.getRequiredChunks(); chunk++) {
+            this._transactionIds.push(nextTransactionId);
+            this._transactionIds.advance();
+
+            if (this._nodeAccountIds.list.length === 0) {
+                this._transactions.push(this._makeSignedTransaction(null));
+            } else {
+                for (const nodeAccountId of this._nodeAccountIds.list) {
+                    this._signedTransactions.push(
+                        this._makeSignedTransaction(nodeAccountId),
+                    );
+                }
+            }
+
+            nextTransactionId = new TransactionId(
+                /** @type {AccountId} */ (nextTransactionId.accountId),
+                new Timestamp(
+                    /** @type {Timestamp} */ (
+                        nextTransactionId.validStart
+                    ).seconds,
+                    /** @type {Timestamp} */ (
+                        nextTransactionId.validStart
+                    ).nanos.add(1),
```

Comment:
Is this adding one nanosecond? Why don't you just generate a transaction id here?

Reply:
Hm, refactored this to generating `TransactionId` - `TransactionId.generate(dummyAccountId);`. You meant like this, right?

Reply:
The dummyAccountId is just a temp value. Why not `TransactionId.generate(nextTransactionId.accountId);` ? Or that won't work for your use case.